### PR TITLE
Fix colum_value length for inserts in table sync_object_field_change_report

### PR DESCRIPTION
**Pull request by julienWebmecanik**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features"
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no

#### Description:
colum_value in sync_object_field_change_report is VARCHAR(191), so trying to insert longer values was creating SQL errors.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Configure any sync Pluggin
3. Try to sync a Contact with a long text field (for example a TEXTAREA message field)
4. The value stored in sync_object_field_change_report will be trimed.

### CODE REVIEW

**Comment by shinde-rahul** [comment]
@julienWebmecanik the `colum_value` is of type `longtext` not `varchar` with 191 length. Here is the how db table looks like,
```
mysql> describe sync_object_field_change_report;
+--------------+---------------------+------+-----+---------+----------------+
| Field        | Type                | Null | Key | Default | Extra          |
+--------------+---------------------+------+-----+---------+----------------+
| id           | int(10) unsigned    | NO   | PRI | NULL    | auto_increment |
| integration  | varchar(191)        | NO   | MUL | NULL    |                |
| object_id    | bigint(20) unsigned | NO   |     | NULL    |                |
| object_type  | varchar(191)        | NO   | MUL | NULL    |                |
| modified_at  | datetime            | NO   |     | NULL    |                |
| column_name  | varchar(191)        | NO   |     | NULL    |                |
| column_type  | varchar(191)        | NO   |     | NULL    |                |
| column_value | longtext            | NO   |     | NULL    |                |
+--------------+---------------------+------+-----+---------+----------------+
8 rows in set (0.00 sec)
```
Please confirm the same.

File: app/bundles/IntegrationsBundle/EventListener/LeadSubscriber.php [hunk 1]
```diff
@@ -199,7 +199,7 @@ private function recordFieldChanges(array $fieldChanges, $objectId, string $obje
                 ->setModifiedAt(new \DateTime())
                 ->setColumnName($key)
                 ->setColumnType($valueDAO->getType())
-                ->setColumnValue($valueDAO->getValue());
+                ->setColumnValue(mb_strlen($valueDAO->getValue()) <= 191 ? $valueDAO->getValue() : substr($valueDAO->getValue(), 0, 191));
```
Comment (shinde-rahul) [inline app/bundles/IntegrationsBundle/EventListener/LeadSubscriber.php:202]: IMO, we should be doing this in the normalization.  And `column_value` is `longtext,` so no need to have this check.

**Review by shinde-rahul — COMMENTED** [review]
Would you please confirm the database table structure of your instance?

File: app/bundles/IntegrationsBundle/EventListener/LeadSubscriber.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/IntegrationsBundle/EventListener/LeadSubscriber.php:202]: @shinde-rahul column_value is varchar 191

image

Comment (kuzmany) [reply app/bundles/IntegrationsBundle/EventListener/LeadSubscriber.php:202]: Probably on your custom branch you have longtext, but in community version there is string. 
@escopecz can you confirm it? We've talking about https://github.com/mautic/mautic/blob/208252c88afa0dd27590b5ae13c8e043651e1010/app/bundles/IntegrationsBundle/Entity/FieldChange.php#L102

Then we should migrate also in community the value to longtext? Can Acquia cherry pick it?

**Comment by escopecz** [comment]
@julienWebmecanik @kuzmany I found the PR that adjusted the field from varchar to longtext: https://github.com/mautic/mautic/pull/10772

I'm closing this one. Please shout if you disagree that #10772 is not a good replacement for this.

**Comment by kuzmany** [comment]
@escopecz did you point correct PR? Because that PR do not cover `sync_object_field_change_report` table

**Comment by escopecz** [comment]
@kuzmany you are right. I went through the commit history and [this commit](https://github.com/mautic/mautic/pull/10772/commits/fc720030ea10b712828b8a581a624a54725da9de) adds the migration and belongs to the PR I shared. How come it's not in the PR itself?

**Comment by escopecz** [comment]
Oh, that's a stupid mistake. The "column_value" keyword is in the migration comment copied from another migration... I'll hunt down the right migration.

**Comment by escopecz** [comment]
Sorry for the brainfart. Here is the correct PR: https://github.com/mautic/mautic/pull/10778

**Pull request closed without merging**